### PR TITLE
Add initial source documentation for Platypus's custom Actions.

### PR DESCRIPTION
**Pull request by alexanderianblair**
Add initial source documentation for Platypus's custom Actions and examples.

### CODE REVIEW

File: doc/content/source/actions/AddFESpaceAction.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## Overview
+
+Action called to add an MFEM finite element space to the problem, parsing content inside an `FESpaces`
+block in the user input. Only has an effect if the `Problem` type is set to `MFEMProblem`.
+
+## Example Input File Syntax
+
+!listing test/tests/kernels/heatconduction.i block=Problem FESpaces
+
+!syntax description /FESpaces/AddFESpaceAction
```
Comment (k-collie) [inline doc/content/source/actions/AddFESpaceAction.md:16]: This line is duplicated from summary, can be removed

**Review by k-collie — CHANGES_REQUESTED** [review]
Could the `addClassDescriptions` in `src/actions` use "an MFEM" rather than "a MFEM"?

In `AddXAction.md` could we link to the docs for `X` in the bit where it says "Action called to add X"?

Look good overall, thanks.

File: doc/content/components.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- [ProblemOperator](source/problem_operators/problem_operator.md): Responsible for updating the
+  state of the system. For transient problems, the `ProblemOperator` is passed to the timestepper.
+
+## Problem Data
+
+- [MFEMMesh](source/mesh/MFEMMesh.md): Builds an `mfem::ParMesh` object from the provided mesh input
+  file.
+
+- [MFEMFECollection](source/fespaces/MFEMFECollection.md): Class specifying a finite element family
+  (set of shape functions) used along with a mesh to define a finite element space.
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/components.md]: Removed (it's built as part of the `MFEMFESpace` - you're right that it's not needed by the end user)

File: doc/content/components.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- [MFEMFESpace](source/fespaces/MFEMFESpace.md): Class responsible for defining the finite element
+  space an `MFEMVariable` is defined with respect to.
+
+- [MFEMVariable](source/variables/MFEMVariable.md): Wrapper for set-up of `mfem::ParGridFunction`
+  objects representing trial variables; stores a set of (true) degrees of freedom for a variable
+  with respect to an `MFEMFESpace`
+
+- [MFEMKernel](source/kernels/MFEMKernel.md/index.md): Class defining integrators contributing to
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/components.md]: Corrected!

File: doc/content/source/actions/AddFESpaceAction.md (hunk 1 above)
Comment (alexanderianblair) [reply doc/content/source/actions/AddFESpaceAction.md]: Removed!

File: doc/content/components.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  the weak form built inside `EquationSystem`. Contains methods returning
+  `mfem::NonlinearFormIntegrators` and `mfem::LinearFormIntegrators` to add to the weak form, along
+  with marker arrays labelling the volumes the (domain) integrators are to be applied to.
+
+- [MFEMBoundaryCondition](source/kernels/MFEMKernel.md/index.md): Class defining essential
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/components.md]: Fixed

**Review by k-collie — APPROVED** [review]

File: doc/content/examples/DefiniteMaxwell.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    \sin(\pi y)\\
+    \sin(\pi z)\\
+    \sin(\pi x)
+\end{pmatrix}
+\end{equation}
+
+In this example, we solve this using the weak form
+
+!equation
+-(\vec\nabla \times \vec E, \vec\nabla \times \vec v)_\Omega + (\vec E, \vec v)_\Omega
```
Comment (k-collie) [inline doc/content/examples/DefiniteMaxwell.md:39]: Is there meant to be a negative sign here? [This random pdf](https://www.math.uci.edu/~chenlong/226/Maxwellweakform.pdf) I found doesn't have one (eq.1 vs eq.10)

File: doc/content/examples/HeatTransfer.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+= (\vec f, \vec v)_\Omega \,\,\, \forall \vec v \in V
+
+where
+
+\begin{equation}
+\begin{split}
+T \in H^1(\Omega) &: T = T_0 \,\,\, \mathrm{on}\,\, \Gamma_\mathrm{D} \\
+v \in H^1(\Omega) &: v = 0 \,\,\, \mathrm{on}\,\, \Gamma_\mathrm{D}
+\end{split}
+\end{equation}
```
Comment (k-collie) [inline doc/content/examples/HeatTransfer.md:36]: Should we repeat the convective heat flux BC here?

**Review by k-collie — COMMENTED** [review]
A few comments on the example docs, if you want these on a different PR I can copy them over.

File: doc/content/examples/HeatTransfer.md (hunk 6 above)
Comment (alexanderianblair) [reply doc/content/examples/HeatTransfer.md:38]: We should include the boundary integration term in the weak form, yes - added

File: doc/content/examples/HeatTransfer.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/examples/HeatTransfer.md:39]: Yes, absolutely - added

File: doc/content/examples/DefiniteMaxwell.md (hunk 5 above)
Comment (alexanderianblair) [reply doc/content/examples/DefiniteMaxwell.md]: Nope - good spot - removed

**Pull request merged**